tests/test_batch.py

```diff
@@ -217,5 +219,335 @@ def given_configuration_file_version_different_as_river_file_version_when_runnin
+            config.set("C1", "Discharge", "1300.0")
+            config.set("C1", "Reference", "1300.0")
+            
+            assert dfastmi.batch.check_configuration(rivers, config) is False
+
+        def given_c_sections_with_incorrect_values_when_check_configuration_then_return_false(self, rivers : RiversObject, config : ConfigParser):
+            self.set_valid_general_section(config, "2.0")
+            self.add_c_section(config, "C1", "1300.0")
+            self.add_c_section(config, "C2", "1300.0")
+            self.add_c_section(config, "C3", "1300.0")
+            self.add_c_section(config, "C4", "1300.0")
+            self.add_c_section(config, "C5", "1300.0")
+            self.add_c_section(config, "C6", "1300.0")
+                
+            assert dfastmi.batch.check_configuration(rivers, config) is False        
+                
+        def given_correct_c_sections_when_check_configuration_then_return_true(self, rivers : RiversObject, config : ConfigParser):
+            self.set_valid_general_section(config, "2.0")
+            self.add_c_section(config, "C1", "1300.0")
+            self.add_c_section(config, "C2", "2000.0")
+            self.add_c_section(config, "C3", "3000.0")
+            self.add_c_section(config, "C4", "4000.0")
+            self.add_c_section(config, "C5", "6000.0")
+            self.add_c_section(config, "C6", "8000.0")
+                
+            assert dfastmi.batch.check_configuration(rivers, config) is True
+        
+        def set_valid_general_section(self, config : ConfigParser, version : str):
+            config.add_section("General")
+            config.set("General", "Version", version)
+            config.set("General", "Branch", "Bovenrijn & Waal")
+            config.set("General", "Reach", 'Bovenrijn                    km  859-867')
+            
+        def add_c_section(self, config: ConfigParser, name: str, value : str):
+            config.add_section(name)
+            config.set(name, "Discharge",   value)
+            config.set(name, "Reference",   value)
+            config.set(name, "WithMeasure", value)
+            
+    class Test_get_levels_v2():
+    
+        @pytest.fixture    
+        def reach(self):
+            reach = ReachAdvanced()
+            reach.hydro_q = [6.7, 8.9, 10.1]
+            reach.autotime = True
+            reach.qfit = [11.11, 12.12]
+            reach.celer_form = 1
+            reach.celer_object = CelerProperties()
+            reach.celer_object.prop_q = [13.13, 14.14]
+            reach.celer_object.prop_c = [15.13, 16.14]
+            return reach
+
+        def given_auto_time_true_when_get_levels_v2_then_return_expected_values(self, reach : ReachAdvanced):
+            reach.qstagnant = 4.5
+            q_threshold = 1.2
+            nwidth = 3.4
+
+            Q, apply_q, time_mi, tstag, T, rsigma, celerity = dfastmi.batch.get_levels_v2(reach, q_threshold, nwidth)
+
+            assert Q == [6.7, 8.9, 10.1]
```

Comment:
should you not also check on length of Q, time, T, etc... to be same as input length of hydro_q

Reply:
It is implicitly tested here, I will add a testcase for that.

Reply:
done